### DISCUSSION TITLE
0.1.0

### DISCUSSION THREAD

**Discussion by gilzoide** [gilzoide/lua-gdextension#37] [Announcements]
Built platforms: Windows, Linux/X11, OSX, iOS, Android

Notes: OSX .dylibs are not signed and will need explicit permission from the user or codesigning for running.
iOS .dylibs are not signed and require codesigning before building the application.

### Added
- `LuaState` class for holding a Lua state and interacting with it.
  You may create as many instances as you want, each one representing an independent Lua state.
- `LuaCoroutine`, `LuaFunction`, `LuaLightUserdata`, `LuaTable` and `LuaUserdata` classes that wrap instances from a Lua state in Godot.
- `LuaError` class that represents errors from Lua code.
- Support for registering `Variant` type in Lua states, so that any Godot data can be manipulated in Lua.
- Support for registering Godot classes in Lua, so you can create instances and access integer constants.
- Support for adding access to Godot singleton objects in Lua, accessible directly by name.
- Support for registering Godot utility functions in Lua, like `print`, `lerp` and `is_same`.
- Support for adding access to Godot global enums in Lua, like `OK`, `TYPE_STRING` and `SIDE_LEFT`.

<hr /><em>This discussion was created from the release <a href='https://github.com/gilzoide/lua-gdextension/releases/tag/0.1.0'>0.1.0</a>.</em>

**Comment by Frityet** [comment]
Thank you so much!!!! I have been waiting for this
